core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -333,5 +342,26 @@ public FormValidation doValidateProxy(
             
             return FormValidation.ok(Messages.ProxyConfiguration_Success());
         }
+
+        private boolean isNoProxyHost(String host, String noProxyHost) {
+            if (host!=null && noProxyHost!=null) {
+                for (Pattern p : getNoProxyHostPatterns(noProxyHost)) {
+                    if (p.matcher(host).matches()) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        private Credentials createCredentials(String userName, String password) {
+            if (userName.indexOf('\\') >= 0){
+                final String domain = userName.substring(0, userName.indexOf('\\'));
+                final String user = userName.substring(userName.indexOf('\\') + 1);
+                return new NTCredentials(user, Secret.fromString(password).getPlainText(), domain, "");
```

Comment:
what happens if you are connecting to a non-NTLM proxy where `\` is a valid character in the username?